src/main/scala/com/microsoft/hyperspace/util/SchemaUtils.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.util
+
+import scala.collection.immutable.ListMap
+
+object SchemaUtils {
+
+  val NESTED_FIELD_PREFIX = "__hs_nested."
+  val NESTED_FIELD_PREFIX_REGEX = "^__hs_nested\\."
+
+  /**
+   * The method prefixes a nested field name that hasn't already been prefixed.
+   * The field name must be nested (it should contain a `.` and its type
+   * should be of [[org.apache.spark.sql.types.StructType]]).
+   *
+   * The inverse operation is [[removePrefixNestedFieldName]].
+   *
+   * @param fieldName The nested field name to prefix.
+   * @return A new prefixed field name.
+   */
+  def prefixNestedFieldName(fieldName: String): String = {
+    if (fieldName.contains(".") && !fieldName.startsWith(NESTED_FIELD_PREFIX)) {
+      s"$NESTED_FIELD_PREFIX$fieldName"
+    } else {
+      fieldName
+    }
+  }
+
+  /**
+   * The method prefixes the nested field names from a map where the keys are
+   * the field names and the values are the nested state of that field.
+   * The field names that are not marked as nested will not be changed.
+   *
+   * See [[prefixNestedFieldName]] method.
+   *
+   * @param fieldNames A [[ListMap]] of field names and nested status.
```

Comment:
Coudl you add some comments about ResolverUtils.resolve?
e.g.
```suggestion
   * The method prefixes the nested field names from a map where the keys are
   * the field names and the values are the nested state of that field
   * which should be the result of [[ResolverUtils.resolve]].
   * The field names that are not marked as nested will not be changed.
   *
   * See [[prefixNestedFieldName]] method.
   *
   * @param fieldNames A list of (filed name, nested status) resolved by [[ResolverUtils.resolve]]
```